transactions/transaction.go

```diff
@@ -118,12 +123,12 @@ func (tx *Transaction) GasPrice() uint64 {
 	return tx.pbData.GasPrice
 }
 
-func (tx *Transaction) GasTipCap() uint64 {
-	return tx.pbData.GasPrice
+func (tx *Transaction) GasTipCap() *big.Int {
+	return tx.pbData.GasFeeTip
```

Comment:
I don't think this will return big.Int, unless you convert the bytes into big.Int. Write the unit test to check that and fix the same.